master/internal/task/allocation.go

```diff
@@ -704,6 +709,7 @@ func (a *Allocation) allNonDaemonsExited() bool {
 
 func (a *Allocation) preempt(ctx *actor.Context) {
 	ctx.Log().Info("decided to gracefully terminate allocation")
+	a.exitReason = fmt.Errorf("preempted for one with higher priority")
```

Comment:
there are more reasons than this an allocation can be preempted and end up here.